mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+| `batch_size` | int | `256` | Batch size used in both training and validation procedure. It specifies the number of indices in each batch. |
+| `sampling_workers` | int | `5` | Number of workers that will cooperate in the sampling procedure in the training and validation. |
+| `last_activation_function` | str | `sigmoid` | Activation function that is applied after the last layer in the model and before the `predictor_type`. Currently, only `sigmoid` is supported. | 
+| `add_reverse_edges` | bool | `False` | Whether the module should add reverse edges for each in the obtained graph. If the source and destination node are of the same type, edges of the same edge type will be created. If the source and destination nodes are different, then the prefix `rev_` will be added to the previous edge type. Reverse edges will be excluded as message passing edges for corresponding supervision edges. |
+
+#### **Output**:
+- `status: bool` -> True if all parameters were successfully updated, False otherwise.
+- `message: str` -> "OK" if all parameters were successfully updated, *Error message* otherwise. 
+
+__You__ can send only the parameters you want to change from the default ones, so e.g the method call could look like this:
+```
+CALL link_prediction.set_model_parameters({num_epochs: 100, node_features_property: "features", tr_acc_patience: 8, target_relation: "CITES", batch_size: 256, last_activation_function: "sigmoid", add_reverse_edges: True})
+YIELD status, message
+RETURN status, message;
+```
+
+### **Training**
+The training method doesn't take any parameters, so it is very simple to use. 
+
+#### **Output**:
+- `training_results: List[Dict[str, float]]` -> List of training results through epochs. Model's performance is evaluated every `console_log_freq` epochs. 
+- `validation results: List[Dict[str, float]]` -> List of validation results through epochs. Model's performance is evaluated every `console_log_freq` epochs. 
+
+You can just call
+```
+CALL link_prediction.train() 
+YIELD training_results, validation_results
+RETURN training_results, validation_results;
+```
+and Voilà, training and validation results through epochs are summarized and returned to __you__. 
+
+### **Get training results**
+
+This method is used when the user wants to get performance data obtained from the last training. It is in the same form as a result of calling the training method. If there is no loaded model, the exception will be thrown.
+
+```
+CALL link_prediction.get_training_results()
+YIELD training_results, validation_results;
+RETURN training_results, validation_results;
+```
+
+#### **Output:**
+- `training_results: List[Dict[str, float]]` -> List of training results through epochs. Model's performance is evaluated every `console_log_freq` epochs. 
+- `validation results: List[Dict[str, float]]` -> List of validation results through epochs. Model's performance is evaluated every `console_log_freq` epochs. 
+
+### **Predict**
```

Comment:
You have to make it a bit more clearer what is the method without the whole query ...

SO write the method in the explanation paragraph

Reply:
Do you mean directly naming the method so it is more precise or?

Reply:
I think that methods don't have names, they are always written as code and resemble a function that takes parameters - even when used as a title

predict()
recommend()
load_context()